src/layer/AbstractSentinelHubV3Layer.ts

```diff
@@ -235,4 +250,51 @@ export class AbstractSentinelHubV3Layer extends AbstractLayer {
     const response = await axios.post(this.dataset.findDatesUTCUrl, payload);
     return response.data.map((date: string) => moment.utc(date).toDate());
   }
+
+  public async getStats(params: GetStatsParams): Promise<GetStats> {
+    if (!params.geometry) {
+      throw new Error('Parameter "geometry" needs to be provided');
+    }
+    if (!params.crs) {
+      throw new Error('Parameter "crs" needs to be provided');
+    }
+    if (!params.resolution) {
+      throw new Error('Parameter "resolution" needs to be provided');
+    }
+    if (!params.fromTime || !params.toTime) {
+      throw new Error('Parameters "fromTime" and "toTime" needs to be provided');
+    }
+
+    const payload: FisPayload = {
+      layer: this.layerId,
+      crs: params.crs.authId,
+      geometry: WKT.convert(params.geometry),
+      time: `${moment.utc(params.fromTime).format('YYYY-MM-DDTHH:mm:ss') + 'Z'}/${moment
+        .utc(params.toTime)
+        .format('YYYY-MM-DDTHH:mm:ss') + 'Z'}`,
+      resolution: undefined,
+      bins: params.bins || 5,
+      type: HistogramType.EQUALFREQUENCY,
+      ...this.getStatsAdditionalParameters(),
+    };
+    // When using CRS=EPSG:4326 one has to add the "m" suffix to enforce resolution in meters per pixel
+    if (params.crs.authId === CRS_EPSG4326.authId) {
+      payload.resolution = params.resolution + 'm';
+    } else {
+      payload.resolution = params.resolution;
+    }
+    if (this.evalscript) {
+      payload.evalscript = btoa(this.evalscript);
```

Comment:
Nice, except - `btoa` is not defined in node.js. This should be:
```
  if (evalscript) {
    if (typeof window !== 'undefined' && window.btoa) {
      payload.evalscript = btoa(evalscript);
    } else {
      payload.evalscript = Buffer.from(evalscript, 'utf8').toString('base64');
    }
  }
```

Alternatively, we could define a new utility function (`agnosticBtoa` or just `btoa` - but then we need to make sure we use the correct one).

EDIT: the same above, in `AbstractSentinelHubV1OrV2Layer`

Reply:
True thanks, this has been fixed